util.go

```diff
@@ -182,3 +185,14 @@ func getBinaryFileName(filePath string, dumpType configureType) string {
 
 	return path.Join(filePath, type2name[dumpType]+"."+binarySuffix)
 }
+
+// TryLock is a try lock implementation
+const mutexLocked = 1 << iota
+
+type TryLock struct {
+	sync.Mutex
+}
+
+func (m *TryLock) TryLock() bool {
```

Comment:
What's your opinion about `TryLock`?  @cch123

Reply:
I copy it from this [article](https://colobu.com/2017/03/09/implement-TryLock-in-Go/) written by dalao @smallnest  I write some unit tests for it. Please correct me if I am wrong.

Reply:
haha, I'm sorry that I'm not so sure about this, so ask another dalao look at it together.

Reply:
> I copy it from this [article](https://colobu.com/2017/03/09/implement-TryLock-in-Go/) written by dalao @smallnest I write some unit tests for it. Please correct me if I am wrong.

Go1.18 Mutex/RWMutex have added the TryLock method, or you can copy it as the implementation. It is simpler.

Reply:
@smallnest thank you~

Reply:
I have modify it's implement based on channel.